From f244ad175718c348c6d03f852b2d8bdf55b0da63 Mon Sep 17 00:00:00 2001
From: EgorBo <egorbo@gmail.com>
Date: Wed, 23 Sep 2015 19:06:37 +0300
Subject: [PATCH] Add MonoHandleRefCountedDelete

---
 Source/Urho3D/Container/RefCounted.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Source/Urho3D/Container/RefCounted.cpp b/Source/Urho3D/Container/RefCounted.cpp
index 61796bf..53949ae 100644
--- a/Source/Urho3D/Container/RefCounted.cpp
+++ b/Source/Urho3D/Container/RefCounted.cpp
@@ -35,13 +35,14 @@ RefCounted::RefCounted() :
     // Hold a weak ref to self to avoid possible double delete of the refcount
     (refCount_->weakRefs_)++;
 }
+extern "C" void MonoHandleRefCountedDelete(void* ptr);
 
 RefCounted::~RefCounted()
 {
     assert(refCount_);
     assert(refCount_->refs_ == 0);
     assert(refCount_->weakRefs_ > 0);
-
+	MonoHandleRefCountedDelete(this);
     // Mark object as expired, release the self weak ref and delete the refcount if no other weak refs exist
     refCount_->refs_ = -1;
     (refCount_->weakRefs_)--;
-- 
1.9.5.msysgit.1

